crates/eips/src/eip6110.rs

```diff
@@ -1,12 +1,79 @@
-//! Contains Deposit request constants, first introduced in the [Prague hardfork](https://github.com/ethereum/execution-apis/blob/main/src/engine/prague.md).
+//! Contains Deposit request types, first introduced in the [Prague hardfork](https://github.com/ethereum/execution-apis/blob/main/src/engine/prague.md).
 //!
 //! See also [EIP-6110](https://eips.ethereum.org/EIPS/eip-6110): Supply validator deposits on chain
 
-use alloy_primitives::{address, Address};
+use alloy_primitives::{address, Address, FixedBytes, B256};
+#[cfg(feature = "serde")]
+use cfg_eval::cfg_eval;
```

Comment:
why do we need this?
can we do without? not familiar with this

Reply:
iiuc we need this because of how `serde_as` works. we could solve this with a helper like `serde(with = "from_str")` I guess

similarly to https://github.com/foundry-rs/foundry/blob/b4a47336038cd4d1342eedb3f9555772585e745f/crates/config/src/lib.rs#L2514-L2534 but without lowercase

Reply:
Its necessary to make `#[serde_as]` conditional on the `serde` feature: https://docs.rs/serde_with/latest/serde_with/guide/serde_as/index.html#gating-serde_as-on-features

Without it we get compilation failures running e.g. `cargo +nightly check --no-default-features --target riscv32imac-unknown-none-elf --manifest-path crates/consensus/Cargo.toml` in the no_std GHA.

Reply:
FWIW @klkvr mentioned in telegram that we could move from `serde_as` to `serde_with` to remove the dep but IMO thats outside the scope of this PR.

Reply:
Ah we cross-posted so I didn't see the foundry link, I'll take a stab at this later this afternoon, thanks!

Reply:
Ok, I added a new `alloy_serde::ssz::json::uint` (de)serializer that if we're happy with I can switch other places to start using as well (in a follow-on PR).  This let me get rid of `cfg_eval` and stop using `serde_with` in `alloy-eips`.

```rust
use alloy_serde;
use serde::{Deserialize, Serialize};

#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
pub struct Container {
    #[serde(with = "alloy_serde::ssz::json::uint")]
    value: u64,
}

let val = Container { value: 18112749083033600 };
let s = serde_json::to_string(&val).unwrap();
 assert_eq!(s, "{\"value\":\"18112749083033600\"}");

let deserialized: Container = serde_json::from_str(&s).unwrap();
assert_eq!(val, deserialized);
```